homeassistant/components/eight_sleep/__init__.py

```diff
@@ -168,9 +189,27 @@ def __init__(
 
         mapped_name = NAME_MAP.get(sensor, sensor.replace("_", " ").title())
         if self._user_obj is not None:
-            mapped_name = f"{self._user_obj.side.title()} {mapped_name}"
-
-        self._attr_name = f"Eight {mapped_name}"
+            self._attr_name = (
+                f"{self._user_obj.user_profile['firstName']}'s {mapped_name}"
+            )
+        else:
+            self._attr_name = f"Eight Sleep {mapped_name}"
         self._attr_unique_id = (
             f"{_get_device_unique_id(eight, self._user_obj)}.{sensor}"
         )
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, _get_device_unique_id(eight, self._user_obj))}
+        )
+
+    async def async_heat_set(self, target: int, duration: int) -> None:
+        """Handle eight sleep service calls."""
+        if self._user_obj is None:
+            raise HomeAssistantError(
+                "This entity does not support the heat set service."
+            )
+
+        await self._user_obj.set_heating_level(target, duration)
+        config_entry_data: EightSleepConfigEntryData = self.hass.data[DOMAIN][
+            self._config_entry.entry_id
+        ]
+        await config_entry_data.heat_coordinator.async_request_refresh()
```

Comment:
Usually we save a reference to the coordinator in the constructor instead.

Reply:
The reason it is done this way is because there are two coordinators, and we can't guarantee that the coordinator that's stored in the constructor is the one we need to refresh - this allows users to target any of the entities on a given bed side for this service

Reply:
Since they need to know about each other, you could pass in your new dataclass to the entity constructor instead.

Reply:
hmm that would require some pretty significant refactoring. Right now the logic expects one coordinator instance and one API instance so the base entity class can pass it to the CoordinatorEntity initialization. I'll have to think on this one but I think that should be a separate change if we do move forward with it

Reply:
This problem is a sign that our design is inconsistent. If we our entity design doesn't fit the service the service shouldn't be on the entity.

Reply:
That's fair, but pre-config flow, this was already using entity_id as an input, so making this change would be another breaking change. Should we still move forward with that for this upcoming release or plan to do that in a future release?

Reply:
If we have a good solution, it's good to combine the breaking changes in the same release.